### PR TITLE
MM-36776 - Initial implementation of Chimera proxy

### PR DESCRIPTION
**Pull request by Szymongib**
#### Summary
This PR is the initial implementation of Chimera Proxy together with the unit tests. It includes support for 4 OAuth providers:
- GitHub
- Zoom
- GitLab
- Microsoft (not fully tested)

The proxy was implemented in a way to support multiple applications for a single OAuth provider in case we want to expand its functionality further, to possibly work together with the Apps framework.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-36776

### CODE REVIEW

**Review by jwilander — APPROVED** [review]
Looks good 👍

**Review by gabrieljackson — APPROVED** [review]
Very clean! Nice work on this.

You did some fancy stuff with setting up the CLI that I plan to steal for some of our other services. ;)

File: cmd/chimera/options.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"github.com/spf13/viper"
+)
+
+type ServerOptions struct {
+	Addr          string
+	AppsConfig    string `mapstructure:"apps-config"`
```
Comment (gigawhitlocks) [inline cmd/chimera/options.go:9]: I'm just curious what this `mapstructure` annotation is? I've never seen that before

**Review by gigawhitlocks — APPROVED** [review]

File: cmd/chimera/options.go (hunk 1 above)
Comment (Szymongib) [reply cmd/chimera/options.go:9]: It is for the https://github.com/mitchellh/mapstructure lib. It is used by Viper. Thanks to it we can map options using `Unmarshal` https://github.com/mattermost/chimera/pull/1/files#diff-7ea14ef053eb21da964d22ba8c857fc1be18c314327726c6d326e949b6d5c633R20

**Pull request merged**